app/uk/gov/hmrc/cgtpropertydisposals/repositories/enrolments/TaxEnrolmentRepository.scala

```diff
@@ -109,11 +106,24 @@ class DefaultTaxEnrolmentRepository @Inject() (mongo: ReactiveMongoComponent)(im
   ): EitherT[Future, Error, Option[TaxEnrolmentRequest]] =
     EitherT[Future, Error, Option[TaxEnrolmentRequest]](
       preservingMdc {
-        findAndUpdate(
-          Json.obj("ggCredId" -> ggCredId),
-          Json.obj("$set"     -> Json.toJson(cgtEnrolmentRequest)),
-          fetchNewObject = true
-        ).map(dbResult => Right(dbResult.result[TaxEnrolmentRequest]))
+        collection
+          .findOneAndUpdate(
+            filter = equal("ggCredId", ggCredId),
+            update = Updates.combine(
+              Updates.set("ggCredId", cgtEnrolmentRequest.ggCredId),
+              Updates.set("cgtReference", cgtEnrolmentRequest.cgtReference),
+              Updates.set("address", Codecs.toBson(cgtEnrolmentRequest.address)),
+              Updates.set("timestamp", Codecs.toBson(cgtEnrolmentRequest.timestamp))
+            ),
+            options = FindOneAndUpdateOptions().returnDocument(ReturnDocument.AFTER)
+          )
+          .toFutureOption()
+          .map { dbResult =>
+            dbResult match {
+              case Some(_) => Right(dbResult)
+              case None    => Left(Error("was not able to update"))
```

Comment:
I think this `case None` is not required because we are using `.recover` to handle this case. But there is no harm putting this code here.